app/components/document-creator.js

```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    tagName: 'document-creator',
+    classNames: 'document-creator',
+    documentTitle: null,
+    isCreating: false,
+
+    actions: {
+      createDocument: function () {
+        this.createDocument();
+      },
+    },
+
+    createDocument: function () {
+      var title = this.get('documentTitle').trim(),
+        component = this;
+
+      this.set('isCreating', true);
+
+      console.log('creating document...');
+      return this.get('documentSource').createManually(title).then(function (doc) {
+        component.sendAction('documentCreated', doc);
+        component.set('isCreating', false);
+      }, function (error) {
+        component.set('isCreating', false);
+      });
+    }
+});
```

Comment:
No reason other than "I didn't think of it at the time". I agree. Your name makes much more sense. I'll change it first chance I get (sometime this afternoon, probably).